pkg/util/service_failure_watcher.go

```diff
@@ -0,0 +1,37 @@
+package util
```

Comment:
Shouldn't be part of the `pkg/util/services` framework? Looks generic enough to be put there.

Reply:
You're right. I moved it and renamed to FailureWatcher (`services.FailureWatcher` in full).